VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -171,6 +172,24 @@ public void onDestroy() {
         EventBus.getDefault().unregister(this);
     }
 
+    @Override
+    public void onRestart() {
+        if (postsSort == DiscussionPostsSort.LAST_ACTIVITY_AT && mSelectedItem != -1) {
+            // Move the last viewed thread to the top
+            discussionPostsAdapter.moveToTop(mSelectedItem);
+            mSelectedItem = -1;
+        }
+
+        nextPage = 1;
+        controller.refresh();
```

Comment:
For now this seems to be a better option, cuz, we can't reload the thread list forcefully. So, silently updating the list ensures a user can always see the newly added posts whenever he/she comes back to this screen.